common/src/main/java/com/brandwatch/minibcr/common/repository/QueryRowMapper.java

```diff
@@ -0,0 +1,21 @@
+package com.brandwatch.minibcr.common.repository;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Component;
+
+import com.brandwatch.minibcr.common.domain.Query;
+
+@Component
+public class QueryRowMapper implements RowMapper {
+
+    @Override
+    public Object mapRow(ResultSet resultSet, int i) throws SQLException {
```

Comment:
```suggestion
    public Query mapRow(ResultSet resultSet, int i) throws SQLException {
```
This is probably why you're needing to cast to a Query, as you're just mapping to an object.

Reply:
Good catch, it was also changing the class declaration to `public class QueryRowMapper implements RowMapper<Query>`